pkg/cli/debug/debug.go

```diff
@@ -594,6 +594,30 @@ func (o *DebugOptions) getContainerImageCommand(pod *corev1.Pod, container *core
 func (o *DebugOptions) transformPodForDebug(annotations map[string]string) (*corev1.Pod, []string) {
 	pod := o.Attach.Pod
 
+	if len(o.ToNamespace) > 0 {
+		svcAcctOrig := pod.Spec.ServiceAccountName
+		volumesOrig := pod.Spec.Volumes
+		clearVolumes(pod)
+		if len(pod.Spec.ServiceAccountName) != 0 {
+			pod.Spec.ServiceAccountName = ""
+		}
+		if len(pod.Spec.DeprecatedServiceAccount) != 0 {
+			pod.Spec.DeprecatedServiceAccount = ""
+		}
+		if len(pod.Spec.InitContainers) != 0 {
+			klog.V(4).Infof("PodSpec InitContainers cleared to run with '--to-namespace', run without '--to-namespace' to keep InitContainers")
+			pod.Spec.InitContainers = nil
+		}
+		if svcAcctOrig != pod.Spec.ServiceAccountName {
+			klog.V(4).Infof("PodSpec ServiceAccountName was cleared to run with '--to-namespace', run without '--to-namespace' to keep ServiceAccountName")
```

Comment:
Make it more explicit `fmt.Fprintf(o.ErrOut, "ServiceAccountName was cleared to run with...`

Reply:
got it